evm/vm/transaction_context.py

```diff
@@ -11,12 +13,18 @@ class BaseTransactionContext:
     """
     _gas_price = None
     _origin = None
+    _log_counter = None
 
     def __init__(self, gas_price, origin):
         validate_uint256(gas_price, title="TransactionContext.gas_price")
         self._gas_price = gas_price
         validate_canonical_address(origin, title="TransactionContext.origin")
         self._origin = origin
+        self._log_counter = itertools.count()
+
+    @property
```

Comment:
The side-effect nature of this property doesn't sit well with me.  Can we just make this a method `get_next_log_idx` or `get_next_log_counter`